source/UI.cpp

```diff
@@ -240,7 +240,7 @@ bool UI::IsDone() const
 
 
 
-// Check if it is time to quit.
+// Check if there are no panels left.
```

Comment:
You might want to combine these 2 comments into one. No panels being left kinda implies that there's nothing left to do, but it might be better to explicitly state that.

Reply:
The usage of 2 stacks (menuPanels and gamePanels) doesn't automatically make empty() always match quit(). Added some more text to make usage of this function more clear.